### DISCUSSION TITLE
Dynamic page switching, possible ?

### DISCUSSION THREAD

**Discussion by udf2457** [vercel/next.js#28376] [Help]
Let's say I've got two-types of page on my site:

- List (i.e. rendered list of items, e.g. news, blog posts, staff lists)
- Marketing (i.e. fixed layout)

Lets's say each page has a unique ID (used by the database to pull the data items for that page) and I want to be able to provide a single generic  "permalink"-style endpoint, e.g.:

- http://example.com/permalink/f95f28c2-11fd-4856-8a28-d335b5d7550f

The database (and by extension a next.js API route) knows what type of page layout should be rendered by that ID (e.g.  that http://example.com/permalink/f95f28c2-11fd-4856-8a28-d335b5d7550f is actually "list" type and therefore rendered by `pages/list.js` or whatever ....).

But I'm not clear how I can do that dynamic switch between pages ? Or even if it is possible at all ?

If it makes any difference, my pages use the `getStaticProps()` method and also make use of Per-Page Layouts (`getLayout()`).

**Comment by oliverloops** [comment]
Hi @udf2457, I understand you want to define a single generic permalink inside one of this dynamic rendered pages, right? (Please correct me if I'm wrong) . Next.js is able to generate a dynamic route and populate with the exact same data provided to that specific page id, but this dynamic routing system can change all the time depending on the number of id's you're passing into. I mean if you define a dynamic route this will be always a dynamic route and that's why you have the option tho create a static or dynamic route from the beginning but isn't possible to switch between both once you define it.

**Comment by udf2457** [comment]
@oliverloops 

Not quite.  Perhaps if I put it another way....

In the database pages have a page ID.

Let's say we have a news page with ID `N1`.

Let's say the "normal" way to access the news page would be via, say, `pages/list/[id].js` (i.e. `pages/list/N1` in the browser).

What I would like to be able to do is have a generic route, say `pages/permalink/[id].js`

So, for example, given `pages/permalink/N1`, we could then select (with the help of the database)  the correct layout (i.e. `pages/list/[id].js`)

Hope this helps clarify ?

**Reply by oliverloops** [reply]
Ok, I understand better. You could put inside your `pages` directory all subdirectories you need with the required permalink name you want. for example, this structure: `pages/permalink/N1` can be built into your project if you create a new folder inside pages directory named as you want (this would be your permalink). e.g: `permalink/[id].js` is a directory living inside pages. This way Next.js keeps this static route above all your dynamic generated files. At least this simulates this required permalink.
